### PR TITLE
Refactor: Update form validation

### PR DESCRIPTION
**Pull request by saashapina**
- update submit form validation with consistent validation styles

### Testing
- `Is your apprenticeship a paid opportunity` field
    - **yes** is required: returns "Your apprenticeship must be paid"
- `Company` field
    - field is required: returns "Please enter your company name"
- `Company website` field
    - field is required: returns "Please enter company website"
    - field must be a url: returns "Please enter a valid URL"
- `Category` field
    - field is required: returns "Please select a category"
- `Duration` field
     - fied is required: returns "Please enter a duration"
     - must be greater than 0: returns "Please enter a value greater than 0"
     - Please enter a value
- `Summary` field
    - field is requred: returns "Please enter a summary of the apprenticeship"
- `Location` field
    - field is required: returns "Please enter a location"
- `Application Link`
    - field is required: returns "Please enter a link to the application"
    - field must be a url: returns "Please enter a valid URL"
- `Past apprentice email`
    - field is required: returns "Please enter a past apprentice email"
    - field must be an email: returns "Please enter a valid email"
- `Email`
    - field is required: returns "Please enter an email address"
    - field must be an email: returns "Please enter a valid email address"

### CODE REVIEW

File: source/templates/layouts/default.hbs [hunk 1]
```diff
@@ -38,6 +38,7 @@ footer:
     {{> _icons }}
     <script src="https://cdnjs.cloudflare.com/ajax/libs/modernizr/2.8.3/modernizr.min.js"></script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-validate/1.19.1/jquery.validate.min.js"></script>
```
Comment (changangus) [inline source/templates/layouts/default.hbs:41]: Never used this Jquery plugin before, but it works great!

File: public/assets/js/scripts.js [hunk 2]
```diff
@@ -1,38 +1,63 @@
...
+        required: true,
+        email: true
+      }
+    },
+    messages: {
+      compensation: "Your apprenticeship must be paid",
+      name: "Please enter your company name",
+      url: {
+        required: "Please enter your company website",
+        url: "Please enter a valid URL"
```
Comment (changangus) [inline public/assets/js/scripts.js:35]: For the URL I think we should explicitly mention that all URLs need the `http:/` prefix, its slightly confusing when a user enters a valid URL but without the prefix and it still throws the error.

File: source/scss/modules/_form.scss [hunk 3]
```diff
@@ -129,17 +129,23 @@ textarea {
...
-  color: $c--shady-lady;
-  margin-top: 5px;
-  &.hidden {
-    display: none;
-  }
-  &--red {
-    color: $c--rust;
-    border-color: $c--rust;
-  }
+.alert,
```
Comment (changangus) [inline source/scss/modules/_form.scss:132]: All they styles look great and it's so much better when everything is consistent! Nice job!

**Review by changangus — CHANGES_REQUESTED** [review]
Hey SaaSha, everything looks really awesome! I only had one nitpick change request for the URL validation feedback. Great job!

File: public/assets/js/scripts.js (hunk 2 above)
Comment (saashapina) [reply public/assets/js/scripts.js:35]: good call!

**Comment by jonoliver** [comment]
👏  This is awesome @saashapina!

The UI behavior is much more consistent and jquery validate makes the configuration easier! 🎉 

Two minor things I noticed:
1.  when the validation applies to an invalid url, the input value changes to upper case. Not sure if that's intended, but it seems a bit jarring IMO
2. Invalid inputs appear nearly identical to error messages, making it difficult to visually discern between them. Maybe tweaking the error message styling a bit would help?
image

**Review by changangus — APPROVED** [review]
Hey SaaSha, I ran the cypress tests and it looks like they cover all the form validations. Great Job on this PR!

**Pull request merged**